app/services/canvas_service.py

```diff
@@ -0,0 +1,161 @@
+        })
+
+    async def get_courses(self):
+        try:
+            url = f"{settings.CANVAS_API_URL}/courses"
+            response = self.session.get(url)
+            response.raise_for_status()
+            return response.json()
+        
+        except requests.RequestException as e:
+            raise LMSNoCourseFetchedException()
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=str(e)
+        )
+
+    async def get_course(self, additional_params=None):
+        try:
+            url = f"{settings.CANVAS_API_URL}/courses/{settings.CANVAS_COURSE_ID}"
+            
+            if additional_params:
+                # Append additional parameters to the URL
+                url += "?" + "&".join([f"{key}={value}" for key, value in additional_params.items()])
+            response = self.session.get(url)
+            response.raise_for_status()
+            return response.json()
+        
+        except requests.RequestException as e:
+            raise LMSNoCourseFetchedException()
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=str(e))
+    
+    # returns a dictionary of assignments for a course
+    async def get_assignments(self):
+        try:
+            url = f"{settings.CANVAS_API_URL}/courses/{settings.CANVAS_COURSE_ID}/assignments"
+            response = self.session.get(url)
+            response.raise_for_status()
+            return response.json()
+        
+        except requests.RequestException as e:
+            raise LMSNoAssignmentFetchedException()
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=str(e))
+
+    async def get_assignment(self, assignment_id):
+        try:
+            url = f"{settings.CANVAS_API_URL}/courses/{settings.CANVAS_COURSE_ID}/assignments/{assignment_id}"
+            response = self.session.get(url)
+            response.raise_for_status()
+            return response.json()
+        
+        except requests.RequestException as e:
+            raise LMSNoAssignmentFetchedException()
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=str(e))
+
+    async def get_users(self, additional_params=None):
+        try:
+            url = f"{settings.CANVAS_API_URL}/courses/{settings.CANVAS_COURSE_ID}/users"
+
+            if additional_params:
+                # Append additional parameters to the URL
+                url += "?" + "&".join([f"{key}={value}" for key, value in additional_params.items()])
+
+            response = self.session.get(url)
+            response.raise_for_status()
+            return response.json()
+        
+        except requests.RequestException as e:
+            raise LMSNoStudentsFetchedException()
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=str(e))
+
+    async def upload_grades(self, assignment_id: int, user_id: int, grade: float):
```

Comment:
calling this upload_grades is kinda misleading considering it is uploading a single grade

Reply:
done